components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/dto/SAMLSSOAuthnReqDTO.java

```diff
@@ -486,4 +493,113 @@ public void addIdpAuthenticationContextProperty(String propertyName, Authenticat
+
+        if (authenticationContextClassRefList == null) {
+            return Collections.emptyList();
+        }
+        return Collections.unmodifiableList(authenticationContextClassRefList);
+    }
+
+    /**
+     * Set Authentication Context Class Reference.
+     *
+     * @param authenticationContextClassRefs list of Authentication Context Class Reference
+     */
+    public void setAuthenticationContextClassRefList(List<SAMLAuthenticationContextClassRefDTO>
+                                                             authenticationContextClassRefs) {
+
+        if (authenticationContextClassRefList == null) {
+            authenticationContextClassRefList = authenticationContextClassRefs;
+        } else {
+            authenticationContextClassRefList.addAll(authenticationContextClassRefs);
+        }
+    }
+
+    /**
+     * Add Authentication Context Class Reference.
+     *
+     * @param authenticationContextClassRefs Authentication Context Class Reference
+     */
+    public void addAuthenticationContextClassRef(
+            SAMLAuthenticationContextClassRefDTO authenticationContextClassRefs) {
+
+        if (authenticationContextClassRefList == null) {
+            authenticationContextClassRefList = new ArrayList<>();
+        }
+        authenticationContextClassRefList.add(authenticationContextClassRefs);
+    }
+
+    /**
+     * Get Requested Attributes.
+     *
+     * @return list of requested attributes
+     */
+    public List<ClaimMapping> getRequestedAttributes() {
+
+        return requestedAttributesList;
+    }
+
+    /**
+     * Set Requested Attributes.
+     *
+     * @param requestedAttributes list of requested attributes
+     */
+    public void setRequestedAttributes(List<ClaimMapping> requestedAttributes) {
+
+        if (requestedAttributesList == null) {
+            requestedAttributesList = requestedAttributes;
+        } else {
+            requestedAttributesList.addAll(requestedAttributes);
+        }
+    }
+
+    /**
+     * Get Authentication Context Comparison.
+     *
+     * @return Authentication Context Comparison
+     */
+    public String getRequestedAuthnContextComparison() {
+
+        return requestedAuthnContextComparison;
+    }
+
+    /**
+     * Set Authentication Context Comparison.
+     *
+     * @param authnContextComparison Authentication Context Comparison
+     */
+    public void setRequestedAuthnContextComparison(String authnContextComparison) {
+
+        requestedAuthnContextComparison = authnContextComparison;
+    }
+
+    /**
+     * Get properties.
+     *
+     * @return request properties
+     */
+    public Properties getProperties() {
+
+        if (properties == null) {
+            properties = new Properties();
+        }
+        return properties;
+    }
+
+    /**
+     * Set properties.
+     *
+     * @param properties request properties
+     */
+    public void setProperties(Properties properties) {
+
+        this.properties = properties;
+    }
```

Comment:
I think it would be better to add the capability to set a property or add a property rather than allowing set a bunch of properties.